src/ChunkedStreamDecoder.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace React\HttpClient;
+
+use Evenement\EventEmitterTrait;
+use React\Stream\DuplexStreamInterface;
+use React\Stream\Util;
+
+class ChunkedStreamDecoder
+{
+    const CRLF = "\r\n";
+
+    use EventEmitterTrait;
+
+    /**
+     * @var string
+     */
+    protected $buffer = '';
+
+    /**
+     * @var string
+     */
+    protected $chunkExtension = '';
+
+    /**
+     * @var int
+     */
+    protected $remainingLength = 0;
+
+    /**
+     * @var bool
+     */
+    protected $nextChunkIsLength = true;
+
+    /**
+     * @var DuplexStreamInterface
+     */
+    protected $stream;
+
+    /**
+     * @param DuplexStreamInterface $stream
+     */
+    public function __construct(DuplexStreamInterface $stream)
```

Comment:
This should probably expect a `ReadableStreamInterface` instead? (e.g. https://github.com/clue/php-utf8-react/blob/master/src/Sequencer.php#L21)

For the reference: The `DuplexStreamInterface` extends the `ReadableStreamInterface`, so this should not affect how it works. However, semantically this class only deals with the readable side of a stream, so it makes sense to support read-only streams as well.